src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties' :
+        Object.assign(state, extraData);
+        break;
+
+      case 'clear' :
+        Object.keys(state).forEach(key => {
+          delete state[key];
+        });
```

Comment:
Could be better with [`for ... in` loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in)